newproject.sh

```diff
@@ -0,0 +1,146 @@
+#!/bin/sh
+set -e
+
+# Script inspired by https://gist.github.com/szeidner/613fe4652fc86f083cefa21879d5522b
+
+PROGNAME=$(basename $0)
+WORKING_DIR=$(cd -P -- "$(dirname -- "$0")" && pwd -P)
+OLD_APPNAME="Base Template"
+OLD_TITLE="template"
+OLD_PACKAGE="io.bloco.template"
+
+die() {
+    echo "$PROGNAME: $*" >&2
+    exit 1
+}
+
+usage() {
+    if [ "$*" != "" ] ; then
+        echo "Error: $*"
+    fi
+
+    cat << EOF
+Usage: $PROGNAME --package-name [PACKAGE_NAME] --title [TITLE] --app-name [APP_NAME]
+Rename an Android app and package.
+
+Options:
+-h, --help                         display this usage message and exit
+-p, --package-name [PACKAGE_NAME]  new package name (i.e. com.example.package)
+-t, --title [TITLE]                new app title (i.e. MyApp)
+
+EOF
+
+    exit 1
+}
+
+packagename=""
+title=""
+while [ $# -gt 0 ] ; do
+    case "$1" in
+    -h|--help)
+        usage
+        ;;
+    -p|--package-name)
+        packagename="$2"
+        shift
+        ;;
+    -t|--title)
+        title="$2"
+        shift
+        ;;
+    -*)
+        usage "Unknown option '$1'"
+        ;;
+    *)
+        usage "Too many arguments"
+      ;;
+    esac
+    shift
+done
+
+if [ -z "$packagename" ] ; then
+    usage "No new package provided"
+fi
+
+if [ -z "$title" ] ; then
+    usage "No Title provided"
+fi
+
+# Enforce package name
+regex='^[a-z][a-z0-9_]*(\.[a-z0-9_]+)+[0-9a-z_]$'
+if ! [[ $packagename =~ $regex ]]; then
+    die "Invalid Package Name: $packagename (needs to follow standard pattern {com.example.package})"
+fi
+
+# Enforce title 
+if [[ "$title" =~ \ |\' ]]; then
+    die "Invalid Title: $title (can't have any spaces)"
+fi
```

Comment:
The title should also start with Uppercase, by I'm not sure if it's a strict requirement.

Reply:
If I am not mistaken its not a strict requirement

Reply:
just tested, it's not